AspNetCore-EchoBot-With-State/Startup.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Bot.Builder.BotFramework;
+using Microsoft.Bot.Builder.Core.Extensions;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace AspNetCore_EchoBot_With_State
+{
+    public class Startup
+    {
+        // This method gets called by the runtime. Use this method to add services to the container.
+        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
+        public Startup(IHostingEnvironment env)
+        {
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(env.ContentRootPath)
+                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
+                .AddEnvironmentVariables();
+
+            Configuration = builder.Build();
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddBot<EchoBot>(options =>
+            {
+                options.CredentialProvider = new ConfigurationCredentialProvider(Configuration);                
+                options.Middleware.Add(new CatchExceptionMiddleware<Exception>(async (context, exception) =>
+                {
+                    await context.TraceActivity("EchoBot Exception", exception);
+                    await context.SendActivity("Sorry, it looks like something went wrong!");
+                }));
```

Comment:
I like this of course, but is having exception handling and tracing too much for this sample? I'm torn 50/50. Just thought I'd call it out and see what others might be thinking.

Reply:
I think it adds value here for showing middleware use and the existence of trace activities. They shouldn't be in the console bot, but I think it's ok her IMHO.

Reply:
@cleemullins I a actually going to suggest just adding a few comments around the middleware etc. Seems to be good useful comments in most places but then sparse in a few areas. If we have other samples to focus on those bits though then that's OK.

Reply:
My goal here, and this is up for debate, is that this will *ALSO* be our base template. This code will be the starting point for many bots. 

Having exceptions show up in the emulator makes things generally easier, it seems to me.